### PR TITLE
Act_and_mul support &  add module test for rope

### PR DESCRIPTION
**Pull request by kiddyjinjin**
### PR Category

Operator

### Type of Change

Other

### Summary of changes

1. Defined `GemsRoPE` and `DeepseekRyarkRoPE` as `nn.Module`s. The latter inherits from the former and reuses the `forward` method.  
2. Added tests for both modules, comparing results against the vLLM implementation. Tests pass under `float32` precision. For `float16` and `bfloat16`, the current strict `atol`/`rtol` thresholds cause mismatches, but end-to-end model integration yields consistent behavior.  
3. Introduced an in-place version of the RoPE kernel that updates `query` and `key` directly, aligning with the vLLM implementation.  
4. Added `act_and_mul`-related module and its corresponding module tests.

### Progress

- [ ] Change is properly reviewed (1 reviewer required, 2 recommended).
- [ ] Change is responded to an issue.
- [ ] Change is fully covered by a UT.

### CODE REVIEW

File: src/flag_gems/runtime/backend/_kunlunxin/fused/rotary_embedding.py [hunk 1]
```diff
@@ -98,13 +101,91 @@ def apply_rotary_pos_emb_kernel(
         tl.store(ok_ptr + output_offs, y, mask=mask)
 
 
+@libentry()
+@triton.jit
+def apply_rotary_pos_emb_inplace_kernel(
```
Comment (iclementine) [inline src/flag_gems/runtime/backend/_kunlunxin/fused/rotary_embedding.py:106]: Does this has the same diff as fused/roatry_embedding.py?

File: modules_tests/test_gems_rotary_embedding.py [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+        max_position_embeddings=max_position_embeddings,
+        base=base,
+        device=device,
+        dtype=dtype,
+        rotary_interleaved=rotary_interleaved,
+        scaling_factor=scaling_factor,
+        **kwargs,
+    ).to(dtype=dtype, device=device)
+
+    if has_vllm():
```
Comment (iclementine) [inline modules_tests/test_gems_rotary_embedding.py:145]: I think  we can consider adding vllm as an optional dependency in pyproject.toml.

```python
[project.optional-dependencies]
test = [
    "pytest>=7.1.0",
    "numpy>=1.26",
    "scipy>=1.14",
]
example = [
    "transformers>=4.40.2",
]
```

File: examples/model_with_vllm_test.py [hunk 3]
```diff
@@ -1,16 +1,33 @@
...
+import random
 import time
 
+import numpy as np
+import torch
 from vllm import LLM, SamplingParams
 
 import flag_gems
 
+os.environ["VLLM_ENABLE_MOE_ALIGN_BLOCK_SIZE_TRITON"] = "1"
```
Comment (iclementine) [inline examples/model_with_vllm_test.py:13]: Why is this used?

File: src/flag_gems/runtime/backend/_kunlunxin/fused/rotary_embedding.py (hunk 1 above)
Comment (kiddyjinjin) [reply src/flag_gems/runtime/backend/_kunlunxin/fused/rotary_embedding.py:106]: I only tested the out-of-place version. I'll try the in-place version as well.

File: modules_tests/test_gems_rotary_embedding.py (hunk 2 above)
Comment (kiddyjinjin) [reply modules_tests/test_gems_rotary_embedding.py:145]: nice suggestion

File: examples/model_with_vllm_test.py (hunk 3 above)
Comment (kiddyjinjin) [reply examples/model_with_vllm_test.py:13]: We currently lack a native MOE implementation, so this setting lets us pick a Triton implementation for now.
But I agree, it doesn’t belong in `model_with_vllm_test.py`

File: src/flag_gems/runtime/backend/_kunlunxin/fused/rotary_embedding.py (hunk 1 above)
Comment (iclementine) [reply src/flag_gems/runtime/backend/_kunlunxin/fused/rotary_embedding.py:106]: I mean “is the kernel for klx is identical to that for nvidia"?

File: examples/model_with_vllm_test.py (hunk 3 above)
Comment (iclementine) [reply examples/model_with_vllm_test.py:13]: Is "pick a Triton implementation" necesasary for testing? Or is it optional to use triton implementation?

File: src/flag_gems/runtime/backend/_kunlunxin/fused/rotary_embedding.py (hunk 1 above)
Comment (kiddyjinjin) [reply src/flag_gems/runtime/backend/_kunlunxin/fused/rotary_embedding.py:106]: Yes, the kernel logic is identical. The only difference lies in the KLX version's API, which takes an extra argument named `CloseUnrollControl`, which is used by the KLX Triton compiler (I think).

File: modules_tests/test_gems_activation.py [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    init_seed(42)
+
+    x1 = torch.randn(*shape, dtype=dtype, device=device)
+    x2 = torch.randn(*shape, dtype=dtype, device=device)
+    x_cat = torch.cat([x1, x2], dim=-1)
+
+    target = GemsSiluAndMul()
+    out_test = target(x1, x2)
+
+    if has_vllm():
```
Comment (0x45f) [inline modules_tests/test_gems_activation.py:25]: There will be no accuracy check on CI, right?

Comment (kiddyjinjin) [reply modules_tests/test_gems_activation.py:25]: Currently not in CI. Might be good to discuss if we want to include it going forward.
